vertical-pod-autoscaler/pkg/recommender/main.go

```diff
@@ -118,7 +118,9 @@ func main() {
 	metrics_recommender.Register()
 	metrics_quality.Register()
 
-	useCheckpoints := *storage != "prometheus"
+	usePrometheus := *storage == "prometheus"
+	useNoStorage := *storage == "none"
+	useCheckpoints := (!usePrometheus) && (!useNoStorage)
```

Comment:
The code should quit with an error if the flag value is incorrect, here it would just ignore the incorrect value and proceed with default.

Reply:
done

<img width="1227" alt="image" src="https://github.com/kubernetes/autoscaler/assets/16960194/7e704dad-c7ad-4430-b373-10927907245b">

Reply:
in both cases was following existing behaviour, but agree that the fail fast is a better way of handling it